pandas/tests/frame/test_mutate_columns.py

```diff
@@ -89,10 +89,26 @@ def test_assign_bad(self):
             df.assign(lambda x: x.A)
         with pytest.raises(AttributeError):
             df.assign(C=df.A, D=df.A + df.C)
-        with pytest.raises(KeyError):
-            df.assign(C=lambda df: df.A, D=lambda df: df['A'] + df['C'])
-        with pytest.raises(KeyError):
-            df.assign(C=df.A, D=lambda x: x['A'] + x['C'])
+        if not PY36:
+            with pytest.raises(KeyError):
+                df.assign(C=lambda df: df.A,
+                          D=lambda df: df['A'] + df['C'])
+            with pytest.raises(KeyError):
+                df.assign(C=df.A, D=lambda x: x['A'] + x['C'])
+
+    def test_assign_dependent(self):
+        df = DataFrame({'A': [1, 2], 'B': [3, 4]})
+        if PY36:
```

Comment:
* Reference issue number below function declaration
* You might want to consider decorating with a test-skipper so that we don't run this test on anything below 3.6.